src/platform_impl/linux/x11/util/geometry.rs

```diff
@@ -264,27 +265,27 @@ impl XConnection {
 
     fn get_parent_window(&self, window: ffi::Window) -> Result<ffi::Window, XError> {
         let parent = unsafe {
-            let mut root: ffi::Window = mem::uninitialized();
-            let mut parent: ffi::Window = mem::uninitialized();
+            let mut root = MaybeUninit::uninit();
+            let mut parent = MaybeUninit::uninit();
             let mut children: *mut ffi::Window = ptr::null_mut();
-            let mut nchildren: c_uint = mem::uninitialized();
+            let mut nchildren = MaybeUninit::uninit();
 
             // What's filled into `parent` if `window` is the root window?
             let _status = (self.xlib.XQueryTree)(
                 self.display,
                 window,
-                &mut root,
-                &mut parent,
+                root.as_mut_ptr(),
+                parent.as_mut_ptr(),
                 &mut children,
-                &mut nchildren,
+                nchildren.as_mut_ptr(),
             );
 
             // The list of children isn't used
             if children != ptr::null_mut() {
                 (self.xlib.XFree)(children as *mut _);
             }
 
-            parent
+            parent.assume_init()
```

Comment:
Afaik, `assume_init` should be called on `root` and `nchildren` so that `Drop` is called on them, else we got a memory leak.